Add I$(LINUX_KERNEL_ROOT)/include/generated/uapi to the include Path for ION headers

diff --git a/Makefile_G1G2 b/Makefile_G1G2
index 8280cce..4c9fc34 100755
--- a/Makefile_G1G2
+++ b/Makefile_G1G2
@@ -10,8 +10,9 @@ INCLUDE_HEADERS = -I./decoder_sw -I$(SOURCE_ROOT)/source/inc -I$(SOURCE_ROOT)/so
 
 INCLUDE_HEADERS += -I$(SOURCE_ROOT)/linux/memalloc
 #INCLUDE_HEADERS += -I$(SOURCE_ROOT)/linux/ldriver
-#INCLUDE_HEADERS += -I$(LINUX_KERNEL_ROOT)/include/uapi -I$(LINUX_KERNEL_ROOT)/include
-INCLUDE_HEADERS += -I$(SDKTARGETSYSROOT)/usr/include/imx
+INCLUDE_HEADERS += -I$(LINUX_KERNEL_ROOT)/include/uapi -I$(LINUX_KERNEL_ROOT)/include
+INCLUDE_HEADERS += -I$(LINUX_KERNEL_ROOT)/drivers/staging/android/uapi
+INCLUDE_HEADERS += -I$(LINUX_KERNEL_ROOT)/include/generated/uapi

 CFLAGS += -DDEC_MODULE_PATH=\"/dev/mxc_hantro\" -DUSE_FAKE_RFC_TABLE -DFIFO_DATATYPE=void* -DNDEBUG -DDOWN_SCALER \
            -DUSE_EXTERNAL_BUFFER -DUSE_FAST_EC -DUSE_VP9_EC -DGET_FREE_BUFFER_NON_BLOCK \
diff --git a/Makefile_H1 b/Makefile_H1
index 9ab62e1..8d366e3 100755
--- a/Makefile_H1
+++ b/Makefile_H1
@@ -17,8 +17,9 @@ ENV += -DMEMALLOC_MODULE_PATH=\\\"/dev/ion\\\"
 #ENV += -DSDRAM_LM_BASE=0x00000000
 ENV += -DEWL_NO_HW_TIMEOUT
 ENV += -DUSE_ION
-#ENV += -I$(LINUX_KERNEL_ROOT)/include/uapi -I$(LINUX_KERNEL_ROOT)/include
-ENV += -I$(SDKTARGETSYSROOT)/usr/include/imx
+ENV += -I$(LINUX_KERNEL_ROOT)/include/uapi -I$(LINUX_KERNEL_ROOT)/include
+ENV += -I$(LINUX_KERNEL_ROOT)/drivers/staging/android/uapi
+ENV += -I$(LINUX_KERNEL_ROOT)/include/generated/uapi 

 LIBENCNAME = libcodec_enc
 LIBSENC = -L./ -lhantro_h1 -lpthread
diff --git a/decoder_sw/software/linux/dwl/dwl_linux.c b/decoder_sw/software/linux/dwl/dwl_linux.c
index 9b057f0..43225f6 100755
--- a/decoder_sw/software/linux/dwl/dwl_linux.c
+++ b/decoder_sw/software/linux/dwl/dwl_linux.c
@@ -41,7 +41,7 @@
 #include "dwl.h"
 #include <linux/hantrodec.h>
 #ifdef USE_ION
-#include <linux/ion.h>
+#include <ion.h>
 #include <linux/dma-buf.h>
 #include <linux/version.h>
 #ifdef ANDROID
diff --git a/h1_encoder/software/linux_reference/ewl/ewl_x280_common.c b/h1_encoder/software/linux_reference/ewl/ewl_x280_common.c
index fef1307..20b6f95 100755
--- a/h1_encoder/software/linux_reference/ewl/ewl_x280_common.c
+++ b/h1_encoder/software/linux_reference/ewl/ewl_x280_common.c
@@ -49,7 +49,7 @@
 
 #include "linux/hx280enc.h"
 #ifdef USE_ION
-#include <linux/ion.h>
+#include <ion.h>
 #include <linux/dma-buf.h>
 #include <linux/version.h>
 #ifdef ANDROID
-- 
2.25.1

